src/main/java/net/dv8tion/jda/core/entities/EntityBuilder.java

```diff
@@ -874,6 +877,42 @@ public Message createMessage(JSONObject jsonObject, MessageChannel chan, boolean
         final List<MessageEmbed>       embeds      = map(jsonObject, "embeds",      this::createMessageEmbed);
         final List<MessageReaction>    reactions   = map(jsonObject, "reactions",   (obj) -> createMessageReaction(chan, id, obj));
 
+        if (!jsonObject.isNull("activity"))
+        {
+            JSONObject activityData = jsonObject.getJSONObject("activity");
+            Message.ActivityType activityType = Message.ActivityType.fromId(activityData.getInt("type"));
+            final String partyId = activityData.getString("party_id");
+            Message.Application application = null;
+
+            switch (activityType)
+            {
+                case PARTY:
+                    activity = new Message.Activity(activityType, partyId, application);
+                    break;
+                case GAME:
+                    {
+                        if (!jsonObject.isNull("application"))
+                        {
+                            JSONObject applicationData = jsonObject.getJSONObject("application");
+
+                            final String name = applicationData.getString("name");
+                            final String description = applicationData.getString("description");
+                            final String iconId = applicationData.getString("icon");
+                            final String coverId = applicationData.getString("cover_image");
+                            final long applicationId = applicationData.getLong("id");
```

Comment:
Use opts here too

Reply:
Is application present for spotify as well?
What would that be filled in with?

Reply:
No it is not check the example jsons i added yesterday in the PR message ;)

Reply:
I think ops arent required anymore because the docs are done now.